src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -454,10 +454,19 @@ def make_zinc_job(target, input_product_key, output_products, dep_keys):
         # Otherwise, fail it.
         on_success=ivts.update,
         on_failure=ivts.force_invalidate)
+        
+    workflow = rsc_compile_context.workflow
+
+    # Replica of JvmCompile's _record_target_stats logic
+    self.context.run_tracker.report_target_info(
+      self.options_scope,
+      compile_target,
+      ['compile', 'workflow'],
+      workflow.value
```

Comment:
I *think* I spent some time making run tracker JSON reporting support `enum`s such as the target `workflow` directly, without converting to string via `.value`. I think you can do:
```suggestion
      workflow
```
and it would work, and I think that would be more consistent with the other ways we've begun to record these enum-style options.

Reply:
Could you give an example of it working with an `enum` somewhere in the codebase? Basically I want to go from "think" to "know".

Reply:
This doesn't seem like a blocking change.

Reply:
Danny pointed out that `enum` discourages accessing the `value` (although we should probably do more to discourage it, because it is a very common practice in other implementations to rely on the value of an enum for serialization).

Reply:
@wiwa when I say *think* I meant "I was writing this on my phone and couldn't find the link". I wrote the `enum` support and I definitely wrote the json encoding support: see https://github.com/pantsbuild/pants/blob/570fab8d3896619b3005c542ee7e61afbd412e67/src/python/pants/goal/run_tracker.py#L36-L46

As the person who took the time to make json encoding enum options work, I would ask you to please make use of it and record the `enum` instance without extracting the private `.value` field.

As I look at the `enum` docstring: https://github.com/pantsbuild/pants/blob/81993f46b2d862fc7a3a52fb51c15f3e59341736/src/python/pants/util/objects.py#L251 I'm realizing the fact that `.value` is intended to be private isn't mentioned anywhere at all. I initially wanted to name it specifically `.private_value_field`, precisely so this confusion wouldn't occur. I'm going to do that in a followup PR.

Reply:
Also @stuhood:
> it is a very common practice in other implementations to rely on the value of an enum for serialization

That's actually what we're doing here and what I'm trying to ensure by asking for options to be recorded without unwrapping the `enum` (because that's what we already do elsewhere in pants). I'm relying on the `enum` serialization logic that is spread across the `CoercingEncoder`, `CoercingOptionEncoder`, and `RunTrackerOptionEncoder` class (in particular, see `pants.base.hash_utils.CoercingEncoder` for the logic regarding `datatype` and `enum` serialization I'm referring to). We rely on the json serialization of *the enum itself* for caching purposes, as `CoercingEncoder` is used in `stable_json_sha1()`, which is used to implement caching in pants. If the `workflow` is a fingerprintable option, we are relying on the `stable_json_sha1()` of its serialization, and that serialization right now doesn't pull out the `.value`, it uses the `CoercingEncoder` to ensure it *always gets serialized the same way*. The point of an `enum` (or one of them) is to attach some strictness / validation to raw data. In my view, one of the jobs of the `enum` wrapper is to serialize itself reliably, and we have that, just by removing the `.value` here.

Reply:
Okay, thanks, will change. I like the philosophy of `enum` responsibilities. Maybe `value` should be `_value`?

Reply:
It doesn't look like this works: https://gist.github.com/stuhood/9b733f7006e08e94bd3bed350d3ed3d2

Sorry for the trouble @wiwa : can switch it back to using `.value`.

Reply:
I've just pushed a branch at https://github.com/pantsbuild/pants/compare/dc9d100...b0571f84af08 which tries to use the more stable `repr()` for enums, to see if that passes travis. It's difficult to investigate failures in https://travis-ci.org/pantsbuild/pants/jobs/563261042 since the error is in the `ast.literal_eval()` method, which processes the `target_dict` after everything writes to it. The branch is going through travis [here](https://travis-ci.org/pantsbuild/pants/builds/563318062).